raiden/network/proxies/secret_registry.py

```diff
@@ -300,7 +315,14 @@ def is_secret_registered(
             secrethash: SecretHash,
             block_identifier: BlockSpecification,
     ) -> bool:
-        """True if the secret for `secrethash` is registered at `block_identifier`."""
+        """True if the secret for `secrethash` is registered at `block_identifier`.
+
+        Throws NoStateForBlockIdentifier if the given block_identifier
+        is older than the pruning limit
+        """
+        if not self.client.can_query_state_for_block(block_identifier):
+            raise NoStateForBlockIdentifier()
```

Comment:
I missed this in the initial review... i think places where `is_secret_registered` should also handle this exception. A quick search shows me that we use this function in `raiden_service.py` and `message_handler.py`

Reply:
No this is not needed. I have already checked those places. You will notice that they are both called with `'latest'`. As such it's impossible to throw this exception.